telosnetwork/teloscan
# #535 | Transactions Route & Table

# See #535 

## Description
Redesign of transaction table UI & functionality, update of route, and refactor to setup syntax for several components


## Test scenarios
- go to https://deploy-preview-546--teloscan-redesign.netlify.app/txs
    - you should see the transactions table
    - addresses in the from/to column should have a contract icon next to them if they are contracts
- click the eye icon in various rows with different qualities (e.g. is TLOS transfer, is Native deposit/withdraw, action has known name, action has no known name)
    - the tx preview should show and the values displayed should make sense for the row
- close the tx preview
- hover over a method name which has other instances on the table, e.g. "Transfer TLOS"
    - other method badges of the same kind should highlight
- hover over a `to` address which shows in the table more than once
    - other instances of the same address should highlight
- hover over a `from` address which shows in the table more than once
    - other instances of the same address should highlight
- click the copy button next to an address
    - the address should be copied to clipboard
- click the 'age' header
    - time format should switch
- click the 'Txn fee' header
    - the data should switch to show gas price
- try filtering by a block:  https://deploy-preview-546--teloscan-redesign.netlify.app/txs?block=286084888
    - data should be filtered
    - you should see appropriate labels in the page header
- try filtering by an account: https://deploy-preview-546--teloscan-redesign.netlify.app/txs?a=0xF5E8A439C599205C1aB06b535DE46681Aed1007a
    - data should be filtered
    - you should see appropriate labels in the page header
- try filtering by block and account: https://deploy-preview-546--teloscan-redesign.netlify.app/txs?a=0xF5E8A439C599205C1aB06b535DE46681Aed1007a&block=286084888
    - data should be filtered
    - you should see appropriate labels in the page header

## Checklist:


TODO:
- [x] action badge text should not be gray for unknown actions
- [x] change eye icon to use OutlineButton
- [x] change styling of transaction quick view status
- [x] in tx quick view, it should not say "Transfer for 0x0 TLOS from ... to ..." if it wasnt a transfer (i.e. if 0 value transaction), instead show action name
- [x] numerical values in quick view should not be hex
- [x] tx value in table should be formatted with toLocaleString
- [x] action badge should still highlight on hover for unknown actions
- [x] page should not say "for block xxxxx" if not filtered by block; filtering by block should work
- [x] action badge should not show an icon for deposits
- [x] clicking the question mark <th> causes the table to change size
- [x] enable filter by address
